src/scripts/scaling/scalingAction.sh

```diff
@@ -16,10 +14,328 @@ scaling_size=1
+  echo "    cluster_name - WebLogic Cluster Name"
+  echo "    kubernetes_master - Kubernetes master URL, default=https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"
+  echo "    access_token - Service Account Bearer token for authentication and authorization for access to REST Resources"
+  echo "    wls_domain_namespace - Kubernetes name space WebLogic Domain is defined in, default=default"
+  echo "    operator_service_name - WebLogic Operator Service name, default=internal-weblogic-operator-svc"
+  echo "    operator_service_account - Kubernetes Service Account for WebLogic Operator, default=weblogic-operator"
+  echo "    operator_namespace - WebLogic Operator Namespace, default=weblogic-operator"
+  echo "    scaling_size - number of WebLogic server instances by which to scale up or down, default=1"
+  exit 1
+}
+
+# Retrieve WebLogic Operator Service Account Token for Authorization
+function initialize_access_token() {
+  if [ -z "$access_token" ]
+  then
+    access_token=`cat /var/run/secrets/kubernetes.io/serviceaccount/token`
+  fi
+}
+
+function logScalingParameters() {
+  trace "scaling_action: $scaling_action"
+  trace "wls_domain_uid: $wls_domain_uid"
+  trace "wls_cluster_name: $wls_cluster_name"
+  trace "wls_domain_namespace: $wls_domain_namespace"
+  trace "operator_service_name: $operator_service_name"
+  trace "operator_service_account: $operator_service_account"
+  trace "operator_namespace: $operator_namespace"
+  trace "scaling_size: $scaling_size"
+}
+
+# Query WebLogic Operator Service Port
+function get_operator_internal_rest_port() {
+  local STATUS=$(curl \
+    -v \
+    --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+    -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    -X GET $kubernetes_master/api/v1/namespaces/$operator_namespace/services/$operator_service_name/status)
+  if [ $? -ne 0 ]
+  then
+    trace "Failed to retrieve status of $operator_service_name in name space: $operator_namespace"
+    trace "STATUS: $STATUS"
+    exit 1
+  fi
+
+  local port
+  if [ -x "$(command -v jq)" ]; then
+    local extractPortCmd="(.spec.ports[] | select (.name == \"rest\") | .port)"
+    port=$(echo "${STATUS}" | jq "${extractPortCmd}")
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+for i in json.load(sys.stdin)["spec"]["ports"]:
+  if i["name"] == "rest":
+    print(i["port"])
+INPUT
+port=$(echo "${STATUS}" | python cmds-$$.py)
+  fi
+  echo "$port"
+}
+
+# Retrieve the api version of the deployed Custom Resource Domain
+function get_domain_api_version() {
+  # Retrieve Custom Resource Definition for WebLogic domain
+  local CRD=$(curl \
+    -v \
+    --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+    -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    -X GET \
+    $kubernetes_master/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/domains.weblogic.oracle)
```

Comment:
This URL path is wrong for K8s 1.16 and above as CRD's graduated from "v1beta1" to "v1". You could either determine the version of the k8s master or I have another idea. If you curl the "$kubernetes_master/apis" path, you get back a list of all API's supported by the server. This is equivalent to `kubectl api-versions`. The code could pick out the highest version of "weblogic.oracle" returned. For instance, on my system I get back "weblogic.oracle/v7" and "weblogic.oracle/v8".

Reply:
We need to test on k8s 1.16 -- it's likely that I'm wrong that this is broken now as 1.16 probably honors the "v1beta1" path. If so, that will be dropped in some k8s release -- I'll try and figure out which one.

Reply:
I found the answer... Your current code will work until k8s 1.22.

Reply:
"v1beta1" is still honored as I'm running 1.18.12:

bash-4.2$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:18:55Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:11:15Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}